commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Messages/MessagesIntegrationTests.cs

```diff
@@ -0,0 +1,145 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using commercetools.Sdk.Client;
+using commercetools.Sdk.Domain;
+using commercetools.Sdk.Domain.Customers;
+using commercetools.Sdk.Domain.Customers.UpdateActions;
+using commercetools.Sdk.Domain.Messages;
+using commercetools.Sdk.Domain.Messages.Customers;
+using commercetools.Sdk.Domain.Messages.Reviews;
+using commercetools.Sdk.Domain.Query;
+using commercetools.Sdk.Domain.Reviews;
+using Xunit;
+using static commercetools.Sdk.IntegrationTests.Customers.CustomersFixture;
+using static commercetools.Sdk.IntegrationTests.Reviews.ReviewsFixture;
+
+namespace commercetools.Sdk.IntegrationTests.Messages
+{
+    /// <summary>
+    /// Assume that messages enabled in danger zone of current project
+    /// </summary>
+    [Collection("Integration Tests")]
+    public class MessagesIntegrationTests
+    {
+        private readonly IClient client;
+
+        public MessagesIntegrationTests(ServiceProviderFixture serviceProviderFixture)
+        {
+            this.client = serviceProviderFixture.GetService<IClient>();
+        }
+
+        [Fact]
+        public async Task GetMessageById()
+        {
+            //lets create customer first to ensure message created
+            await WithCustomer(client, async customer =>
+            {
+                var queryCommand = new QueryCommand<Message>();
+                queryCommand.SetLimit(1);
+                var returnedSet = await client.ExecuteAsync(queryCommand);
+                Assert.Single(returnedSet.Results);
+                var messageId = returnedSet.Results[0].Id;
+
+                var retrievedMessage = await client
+                    .ExecuteAsync(new GetByIdCommand<Message>(messageId));
+                Assert.NotNull(retrievedMessage);
+            });
+        }
+
+        [Fact]
+        public async Task QuerySpecificTypeOfMessages()
+        {
+            //lets create customer first to ensure message created
+            await WithCustomer(client, async customer =>
```

Comment:
This should be done with assert eventually logic. And I would also add the customer id to the query to ensure to capture the right message directly. Then you also don't need the sorting at all.

Reply:
I did the Assert Eventually but we can't query by customer Id for Messages as it's not common for all messages, so it will give The field 'customer' does not exist Error

Reply:
I meant something like `message.Type == "CustomerCreated" && message.resource.id == customer.Id.valueOf()`

Reply:
I did it